### ISSUE TITLE
Vision

### ISSUE THREAD

**Issue by AndriyKy** [AndriyKy/spot-gazer#27]
## About the project

The work is based on the idea of capturing images through a video camera, transmitting the video stream to a server, analyzing and detecting parking lot congestion using code written in the Python programming language, and transmitting the information to a website with a map of all available parking lots in the city.
The thing about the project is that it can be non-profit, like Wikipedia, there are a lot of drivers, and they all need to park somewhere. 

---

### ✅ | https://github.com/AndriyKy/spot-gazer/issues/23 | [Core] Determination of parking lot congestion

 Once every 5, 10, 30, 60 seconds, a frame is taken from the camera, the parking lot congestion is determined, the data is recorded in the RAM database and then intermittently transferred to the ROM database.

### 🔳 | https://github.com/AndriyKy/spot-gazer/issues/24 | [Core] Obtain the coordinates of parking zones

Get a frame from the video stream. Use `label-studio` to mark parking zones. Export the result as a COCO archive. Convert the archive.

### 🔳 | https://github.com/AndriyKy/spot-gazer/issues/25 | [Backend] Database

The administrator is responsible for entering parking lots into the database (location, capacity, etc.), responding to incorrectly parked cars

### 🔳 | https://github.com/AndriyKy/spot-gazer/issues/26 | [Frontend] Map

Displays a map with all available parking lots

### ✅ | Testing

Cover at least 80% of the code with tests.

## Tools

- **Programming language:** Python
- **Recognition tool: +** [Cloud Vision API](https://cloud.google.com/vision/docs/labels?authuser=1) **|** OpenCV ([1](https://www.folio3.ai/blog/car-detection-using-opencv-and-python-within-5-minutes/), [2](https://www.geeksforgeeks.org/opencv-python-program-vehicle-detection-video-frame/)) **|** CNN (R-CNN) **|** PyTorch **| +** [YOLO](https://docs.ultralytics.com/yolov5/) ([1](https://medium.com/analytics-vidhya/object-detection-using-yolo-and-car-detection-implementation-1ec79e882875), [2](https://youtu.be/fu2tfOV9vbY), [3](https://docs.ultralytics.com/)) **| +** [TensorFlow](https://www.tensorflow.org/) **|** [MobileNet](https://xailient.com/blog/real-time-vehicle-detection-with-mobilenet-ssd-and-xailient/)
- **DB: +** Redis **| +** PostgreSQL **|** MySQL
- **Web-framework:** FastAPI **| +** [Django](https://www.djangoproject.com/) **|** Flask
- **Frontend:** ~~[Google Maps API](https://github.com/googlemaps/google-maps-services-python)~~ **| +** [Folium](https://python-visualization.github.io/folium/) **|** [Leaflet](https://leafletjs.com/index.html)
- **Web hosting:** Apace Tomcat Server
- **Other:** caching, ~~multithreading~~, asynchronous, [Label Studio](https://labelstud.io/)
